src/transformers/models/dallemega/modeling_dallemega.py

```diff
@@ -0,0 +1,1557 @@
+        )
+
+
+@add_start_docstrings(
+    "The DALLE_MEGA Model with a language modeling head. Can be used for summarization.", DALLE_MEGA_START_DOCSTRING
+)
+class DalleMegaForConditionalGeneration(DalleMegaPretrainedModel):
+    base_model_prefix = "model"
+
+    def __init__(self, config: DalleMegaConfig):
+        super().__init__(config)
+        self.model = DalleMegaModel(config)
+        self.lm_head = nn.Linear(config.d_model, config.decoder_vocab_size, bias=False)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_encoder(self):
+        return self.model.get_encoder()
+
+    def get_decoder(self):
+        return self.model.get_decoder()
+
+    def resize_token_embeddings(self, new_num_tokens: Optional[int] = None) -> nn.Embedding:
+        """
+        Resizes input token embeddings matrix of the encoder if `new_num_tokens != config.encoder_vocab_size`.
+
+        Takes care of tying weights embeddings afterwards if the model class has a `tie_weights()` method.
+
+        Arguments:
+            new_num_tokens (`int`, *optional*):
+                The number of new tokens in the encoder embedding matrix. Increasing the size will add newly
+                initialized vectors at the end. Reducing the size will remove vectors from the end. If not provided or
+                `None`, just returns a pointer to the input tokens `torch.nn.Embedding` module of the model without
+                doing anything.
+
+        Return:
+            `torch.nn.Embedding`: Pointer to the input tokens Embeddings Module of the model.
+        """
+        base_model = getattr(self, self.base_model_prefix, self)
+        return base_model.resize_token_embeddings(new_num_tokens)
+
+    def resize_decoder_token_embeddings(self, new_num_tokens):
+        old_embeddings = self.model.get_decoder_input_embeddings()
+        new_embeddings = self._get_resized_embeddings(old_embeddings, new_num_tokens)
+        self.model.set_decoder_input_embeddings(new_embeddings)
+
+        # if word embeddings are not tied, make sure that lm head is resized as well
+        if self.get_output_embeddings() is not None and not self.config.tie_word_embeddings:
+            old_lm_head = self.get_output_embeddings()
+            new_lm_head = self._get_resized_lm_head(old_lm_head, new_num_tokens)
+            self.set_output_embeddings(new_lm_head)
+
+        model_embeds = self.model.get_decoder_input_embeddings()
+
+        if new_num_tokens is None:
+            return model_embeds
+
+        # Update base model and current model config
+        self.config.decoder_vocab_size = new_num_tokens
+
+        # Tie weights again if needed
+        self.tie_weights()
+
+        return model_embeds
+
+    def get_output_embeddings(self):
+        return self.lm_head
+
+    def set_output_embeddings(self, new_embeddings):
+        self.lm_head = new_embeddings
+
+    def tie_weights(self):
```

Comment:
Why is this here? can't this just use the standard `tie_weights` function in `modeling_utils.py`?